cli/tests/activate.bats

```diff
@@ -4135,3 +4135,102 @@ EOF
 
   zsh_repeat_activation_PATH .zshenv.extra .zshrc
 }
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=activate,activate:profile
+@test "profile scripts run for all environments in nested activation" {
+  project_setup
+
+  OUTER_MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+    [profile]
+    common = """
+      echo "$_FLOX_SHELL_FORCE"": common hook running from outer"
+    """
+    bash = """
+      alias outer="echo bash: the outer alias is defined"
+      alias current="echo bash: the current alias is outer"
+    """
+    fish = """
+      alias outer="echo fish: the outer alias is defined"
+      alias current="echo fish: the current alias is outer"
+    """
+    tcsh = """
+      alias outer 'echo tcsh: the outer alias is defined'
+      alias current 'echo tcsh: the current alias is outer'
+    """
+    zsh = """
+      alias outer="echo zsh: the outer alias is defined"
+      alias current="echo zsh: the current alias is outer"
+    """
+EOF
+  )"
+  "$FLOX_BIN" init -d outer
+  echo "$OUTER_MANIFEST_CONTENTS" | "$FLOX_BIN" edit -d outer -f -
+
+  INNER_MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+    [profile]
+    common = """
+      echo "$_FLOX_SHELL_FORCE"": common hook running from inner"
+    """
+    bash = """
+      alias inner="echo bash: the inner alias is defined"
+      alias current="echo bash: the current alias is inner"
+    """
+    fish = """
+      alias inner="echo fish: the inner alias is defined"
+      alias current="echo fish: the current alias is inner"
+    """
+    tcsh = """
+      alias inner 'echo tcsh: the inner alias is defined'
+      alias current 'echo tcsh: the current alias is inner'
+    """
+    zsh = """
+      alias inner="echo zsh: the inner alias is defined"
+      alias current="echo zsh: the current alias is inner"
+    """
+EOF
+  )"
+  "$FLOX_BIN" init -d inner
+  echo "$INNER_MANIFEST_CONTENTS" | "$FLOX_BIN" edit -d inner -f -
+
+  # Use _FLOX_SHELL_FORCE in preference to FLOX_SHELL because the activate
+  # script explicitly undefines FLOX_SHELL in the activated environment.
+  _bash="$(command -v bash)"
+  _FLOX_SHELL_FORCE="$_bash" NO_COLOR=1 run -0 expect "$TESTS_DIR/activate/nested-activate.exp" "$PROJECT_DIR"
```

Comment:
question blocking: do we need coverage for in-place and command mode as well? If not, I think it would be better to test the behavior with command mode and an echo rather than an alias rather than using expect

Reply:
At least for bash, aliases are only enabled in interactive shells, and command mode by definition is not interactive so we would need to find a different method of tracing the profile script nesting. That said, I strongly suspect that the userShell shouldn't be involved in command mode at all, so would rather defer that entire topic until we have a chance to plan the refactor.

As for in-place mode, sure we can do that but that doesn't avoid the use of `expect` and I don't know what it proves that the subshell invocation does not. Nevertheless I've authored the changes in commit 8f5593033a57ff5a807ba8e381fe7c80d0a191c2 and currently confirming that tests pass in CI.

Reply:
> At least for bash, aliases are only enabled in interactive shells, and command mode by definition is not interactive so we would need to find a different method of tracing the profile script nesting.

That's why I suggested using an echo

> As for in-place mode, sure we can do that but that doesn't avoid the use of `expect` and I don't know what it proves that the current subshell invocation does not

If it proves something different, I think we should test every mode. If it doesn't, I think we should test just one mode other than interactive, because expect tests are slow and can be flakey

Reply:
OK commit 2e7d2a3e16f831198d74f99e1ea2f1ee71851bab splits this into 3 tests and adds a test for the command mode using echo. With that I'll resolve the conversation and ping Matthew out of band to reopen if this doesn't do the trick.